doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -126,114 +152,156 @@ Type declarations
 
 One can define new types with the following commands.
 
-.. cmd:: Ltac2 Type {? @ltac2_typeparams } @lident
+.. cmd:: Ltac2 Type {? rec } @tac2typ_def {* with @tac2typ_def }
    :name: Ltac2 Type
 
-   This command defines an abstract type. It has no use for the end user and
-   is dedicated to types representing data coming from the OCaml world.
+   .. insertprodn tac2typ_def tac2rec_field
+
+   .. prodn::
+      tac2typ_def ::= {? @tac2typ_prm } @qualid {? {| := | ::= } @tac2typ_knd }
+      tac2typ_prm ::= @ltac2_typevar
+      | ( {+, @ltac2_typevar } )
+      tac2typ_knd ::= @ltac2_type
+      | [ {? {? %| } {+| @tac2alg_constructor } } ]
+      | [ .. ]
+      | %{ {? {+; @tac2rec_field } {? ; } } %}
+      tac2alg_constructor ::= @ident
+      | @ident ( {*, @ltac2_type } )
+      tac2rec_field ::= {? mutable } @ident : @ltac2_type
+
+   :n:`:=`
+     Defines a type with a manifest
 
-.. cmdv:: Ltac2 Type {? rec} {? @ltac2_typeparams } @lident := @ltac2_typedef
+   :n:`::=`
+     Extends an existing open variant type, a special kind of variant type whose constructors are not
+     statically defined, but can instead be extended dynamically. A typical example
+     is the standard `exn` type for exceptions. Pattern matching on open variants must always
+     include a catch-all clause. They can be extended with this form, in which case
+     :token:`tac2typ_knd` should be in the form :n:`[ {? {? %| } {+| @tac2alg_constructor } } ]`.
 
-   This command defines a type with a manifest. There are four possible
-   kinds of such definitions: alias, variant, record and open variant types.
+     .. todo PR above: does it add an additional constructor for an existing o v type or
+        does it create both the type (and apparently the constructor?
 
-   .. productionlist:: coq
-      ltac2_typedef    : `ltac2_type`
-                       : [ `ltac2_constructordef` | ... | `ltac2_constructordef` ]
-                       : { `ltac2_fielddef` ; ... ; `ltac2_fielddef` }
-                       : [ .. ]
-      ltac2_constructordef   : `uident` [ ( `ltac2_type` , ... , `ltac2_type` ) ]
-      ltac2_fielddef         : [ mutable ] `ident` : `ltac2_type`
+   Without :n:`{| := | ::= }`
+     Defines an abstract type for use representing data from OCaml.  Not for
+     end users.
 
-   Aliases are just a name for a given type expression and are transparently
-   unfoldable to it. They cannot be recursive. The non-terminal
-   :production:`uident` designates identifiers starting with an uppercase.
+   :n:`with @tac2typ_def`
+     Permits definition of mutually recursive type definitions.
+
+   Each production of :token:`tac2typ_knd` defines one of four possible kinds
+   of definitions, respectively: alias, variant, open variant and record types.
+
+   Aliases are names for a given type expression and are transparently
+   unfoldable to it. They cannot be recursive.
```

Comment:
```suggestion
   unfoldable to that expression. They cannot be recursive.
```

Reply:
Resolved because you disagree with the suggestion?

Reply:
Resolved because I changed the document in the "More changes" commit.  I think the "Outdated" tells you that this part of the document has been changed:

![image](https://user-images.githubusercontent.com/1253341/89828748-53318200-db0e-11ea-98df-a4fc0468dd02.png)